components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java

```diff
@@ -47,39 +56,93 @@
 
   private final RepositoryManager repositoryManager;
 
-  private final Set<String> memberNames = Sets.newLinkedHashSet();
+  private final ConfigurationStore configurationStore;
+
+  private final Map<String, Recipe> recipes;
+
+  private Config config;
+  private static class Config
+  {
+    @NotNull
+    public List<String> memberNames;
+
+    @Override
+    public String toString() {
+      return getClass().getSimpleName() + "{" +
+          "memberNames='" + memberNames +
+          '}';
+    }
+  }
 
   @Inject
-  public GroupFacetImpl(final RepositoryManager repositoryManager) {
+  public GroupFacetImpl(final RepositoryManager repositoryManager,
+                        final ConfigurationStore configurationStore, final Map<String, Recipe> recipes)
+  {
     this.repositoryManager = checkNotNull(repositoryManager);
+    this.configurationStore = checkNotNull(configurationStore);
+    this.recipes = checkNotNull(recipes);
   }
 
   // TODO: Check for compatibility and cyclic-references
 
   @Override
   protected void doConfigure() throws Exception {
-    NestedAttributesMap attributes = getRepository().getConfiguration().attributes(CONFIG_KEY);
-    memberNames.addAll(attributes.require("memberNames", COLLECTION_STRING));
-    log.debug("Members names: {}", memberNames);
+    config = facet(ConfigurationFacet.class).readObject(CONFIG_KEY, Config.class);
+
+    Map<String, Configuration> memberConfigs = memberConfigs();
+    for (String memberName : config.memberNames) {
+      ensureMemberExists(memberConfigs, memberName);
+      ensureFormatMatches(memberName, getRepository().getFormat(), memberConfigs);
+    }
+  }
+
+  private void ensureMemberExists(final Map<String, Configuration> memberConfigs, final String memberName) {
+    if (!memberConfigs.containsKey(memberName)) {
+      throw new ValidationException("Group refers to nonexistent member repository " + memberName);
```

Comment:
unsure this is proper, as this will only catch one invalid member, creating a potential whack-a-mole issue to validation.